src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1256,9 +1256,13 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetupSrpHost(co
     VerifyOrExit(aHostName, error = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(strlen(aHostName) < SrpClient::kMaxHostNameSize, error = CHIP_ERROR_INVALID_STRING_LENGTH);
 
-    memcpy(mSrpClient.mHostName, aHostName, strlen(aHostName) + 1);
-    error = MapOpenThreadError(otSrpClientSetHostName(mOTInst, aHostName));
-    SuccessOrExit(error);
+    // Avoid adding the same host name multiple times
+    if (strcmp(mSrpClient.mHostName, aHostName) != 0)
+    {
+        memcpy(mSrpClient.mHostName, aHostName, strlen(aHostName) + 1);
```

Comment:
is this the same as strcpy ?

Reply:
strcpy copies until null character including it, memcpy with strlen(aHostName) + 1 will act the same indeed

Reply:
Why not use strcopy? memcpy always feels a little odd/nervous to see with string manipulation :)

Reply:
That was the intent of my comment: using strcpy looks a lot clearer.

Reply:
Yes, you all are right, I'm going to fix this soon

Reply:
It's done